### ISSUE TITLE
[BUG] Removing the first node from a cluster, and rebuilding it, it will not rejoin the cluster.

### ISSUE THREAD

**Issue by jzmatrix** [harvester/harvester#2470]
**Describe the bug**
After creating a multi-node cluster (3 in this case), and removing the first node from the UI, and then physically rebuilding the node, it will not rejoin the cluster.   The remaining (2) nodes are still functional and accessible, but no new nodes are able to join the remaining cluster.

**To Reproduce**
1. On the first node, create a new cluster.
2. Config 'x' nodes, all joining the initial cluster (waiting for every node to go 'Ready' before adding a new node.
3. Remove the first node from the Harvester UI, leaving the remaining nodes intact and available
4. Wipe (or replace) the drives in the initial load and perform a clean node, requesting for it to join the cluster.  The node will show as 'NotReady' and will not join the cluster.

**Expected behavior**
The reloaded node should successfully join the existing cluster.

**Support bundle**
[01-supportbundle_0b532da3-637c-4b06-8b26-bbc7c63be1f1_2022-07-08T01-58-31Z.zip](https://github.com/harvester/harvester/files/9068656/01-supportbundle_0b532da3-637c-4b06-8b26-bbc7c63be1f1_2022-07-08T01-58-31Z.zip)
- The '01' support-bundle is a baseline, prior to removing the node from the cluster.

[02-supportbundle_0b532da3-637c-4b06-8b26-bbc7c63be1f1_2022-07-08T02-23-51Z.zip](https://github.com/harvester/harvester/files/9068658/02-supportbundle_0b532da3-637c-4b06-8b26-bbc7c63be1f1_2022-07-08T02-23-51Z.zip)
- The '02' support-bundle was generated after removing the node from the cluster.

**Environment**
 - Harvester ISO version:  1.0.2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Whitebox machine, X64, 8C, 32G each node (3)

**Additional context**

**Comment by w13915984028** [comment]
> 4 Wipe (or replace) the drives in the initial load and perform a clean node, requesting for it to join the cluster. The node will show as 'NotReady' and will not join the cluster.
 
 @jzmatrix 
 Could you please describe in details about how step 4 was done,  e.g. What `drives` were wiped, and how to `perform a clean node`. thanks.

**Comment by jzmatrix** [comment]
> > 4 Wipe (or replace) the drives in the initial load and perform a clean node, requesting for it to join the cluster. The node will show as 'NotReady' and will not join the cluster.
> 
> @jzmatrix Could you please describe in details about how step 4 was done, e.g. What `drives` were wiped, and how to `perform a clean node`. thanks.

In one scenario I physically replaced the drive in the original system with a new drive.  In a second scenario, I used GParted to wipe the partition table of the drive in the system to start clean.   In both scenarios, there was only one drive in the system.

**Comment by futuretea** [comment]
@jzmatrix Hi, are you use the vip as the management address during installation?
![image](https://user-images.githubusercontent.com/15064560/181540078-e4d43658-9910-4b2b-86e3-664748f59ed1.png)

**Comment by tgazik** [comment]
I have same problem: https://github.com/harvester/harvester/issues/2665
tested in v1.0.3 and 2 different environments.
I configured the nodes via yaml in installation.
vip: myipaddress
vip_mode: static

**Comment by FrankYang0529** [comment]
This issue can be fixed https://github.com/harvester/harvester-installer/pull/344 and https://github.com/rancher/rancherd/pull/25. Following are my test steps with [vagrant-pxe-harvester](https://github.com/harvester/ipxe-examples).

1. Setup a 4-node cluster (harvester-node-0, harvester-node-1, harvester-node-2, harvester-node-3).
2. After all 4-node cluster is installed, remove harvester-node-3 and harvester-node-0.
```
kubectl delete node harvester-node-3
kubectl delete node harvester-node-0
vagrant destroy harvester-node-3
vagrant destroy harvester-node-0
```
3. Run `vagrant ssh pxe_server` to update some content for joining harvester-node-0.
```
# copy config-join-1.yaml to config-join-0.yaml
cp /var/www/harvester/config-join-1.yaml /var/www/harvester/config-join-0.yaml

# update os.hostname to harvester-node-0 in config-join-0.yaml
vim /var/www/harvester/config-join-0.yaml

# change owner of config-join-0.yaml
chown www-data /var/www/harvester/config-join-0.yaml

# update 02:00:00:0d:62:e2. In the last line, change config-create.yaml to config-join-0.yaml
vim /var/www/harvester/02:00:00:0d:62:e2

# restart nginx
systemctl restart nginx.service
```
4. Run `vagrant up harvester-node-0` to start the node again. It should join the cluster and the status will be `Ready`.
5. Run `vagrant up harvester-node-3` to start the node again. It should join the cluster and the status will be `Ready`.

**Comment by bk201** [comment]
@jzmatrix We'll close this one as it should be duplicated with https://github.com/harvester/harvester/issues/2665. The issue should be addressed in v1.1.0. Thanks for the report.
